web/courses.go

```diff
@@ -242,25 +244,53 @@ func (s *AutograderService) getAllLabs(request *pb.LabRequest) ([]*pb.LabResultL
 
 	// populate the final slice to return to the client
 	allCourseLabs := make([]*pb.LabResultLink, 0)
-	for _, usr := range course.Enrollments {
-		// collect all submission values for the user
-		userSubmissions := make([]*pb.Submission, 0)
-		for _, v := range labCache[usr.GetUserID()] {
-			tmp := v
-			userSubmissions = append(userSubmissions, &tmp)
+
+	if request.GetGroupLabs() {
+		for _, grp := range course.Groups {
```

Comment:
This function start to appear a bit long. Consider breaking up into functions.

Reply:
Done.